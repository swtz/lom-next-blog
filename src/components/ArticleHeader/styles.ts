import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      font-size: ${theme.font.sizes.xxlarge};
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.lteMedium} {
        font-size: ${theme.font.sizes.large};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    margin: ${theme.spacings.medium} 0;

    @media ${theme.media.lteMedium} {
      font-size: 2rem;
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
