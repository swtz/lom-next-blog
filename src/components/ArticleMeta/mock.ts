import { ArticleMetaProps } from '.';

export default {
  createdAt: '2025-02-12T11:41:13.991Z', // value from current Strapi API
  author: {
    data: {
      id: '1',
      attributes: {
        displayName: 'surfista',
        slug: 'surfista',
      },
    },
  },

  categories: {
    data: [
      {
        id: '4',
        attributes: {
          displayName: 'Off Topic',
          slug: 'off-topic',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'tech',
          slug: 'tech',
        },
      },
    ],
  },
} as ArticleMetaProps;
