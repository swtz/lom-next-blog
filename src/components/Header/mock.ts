import { HeaderProps } from '.';

export default {
  blogName: 'Blog do Arroz Ã© vida!',
  blogDescription: `Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa.`,
  logo: '/assets/images/logo.svg',
  showText: true,
} as HeaderProps;
