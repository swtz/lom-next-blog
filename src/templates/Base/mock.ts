import { BaseProps } from '.';
import { data } from '../../api/dados.json';

export default {
  setting: data.setting,
  children: 'Hello, world!',
} as BaseProps;
