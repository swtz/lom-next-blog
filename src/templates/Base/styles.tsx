import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css``}
`;
export const HeaderContainer = styled.div`
  ${() => css``}
`;
export const ContentContainer = styled.div`
  ${() => css``}
`;
export const FooterContainer = styled.div`
  ${() => css``}
`;
