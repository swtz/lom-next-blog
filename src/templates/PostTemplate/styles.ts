import styled, { css } from 'styled-components';

export const TagsContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large};
    width: 100%;
    margin: 0 auto;
    max-width: ${theme.sizes.content};
  `}
`;
