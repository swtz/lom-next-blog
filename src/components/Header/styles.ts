import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    /* padding: ${theme.spacings.xhuge}; */
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.xsmall};
    margin: 0 auto;

    ${HeadingStyles} {
      margin: 0 0 calc(${theme.spacings.xsmall} - 1rem);
    }

    @media ${theme.media.lteXSmall} {
      flex-flow: column wrap;

      > div {
        ${HeadingStyles} {
          margin: ${theme.spacings.xsmall} 0
            calc(${theme.spacings.small} - 1rem);
        }
      }
    }
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
  `}
`;
