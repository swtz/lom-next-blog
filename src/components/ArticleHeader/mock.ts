import { ArticleHeaderProps } from '.';
import { ArticleMetaProps } from '../ArticleMeta';

const metadata = {
  createdAt: '2025-02-12T11:41:13.991Z',
  author: {
    data: {
      id: '1',
      attributes: {
        displayName: 'surfista',
        slug: 'surfista',
      },
    },
  },

  categories: {
    data: [
      {
        id: '4',
        attributes: {
          displayName: 'Off Topic',
          slug: 'off-topic',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'tech',
          slug: 'tech',
        },
      },
    ],
  },
} as ArticleMetaProps;

export default {
  title: 'Hello, rice! With rice is possible make a several delicious dishes',
  excerpt: `Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa (Asian rice).`,
  cover: {
    data: {
      id: 1,
      attributes: {
        url: '/assets/images/arroz-branco-saudavel.webp',
        alternativeText: null,
      },
    },
  },
  metadata: metadata,
} as ArticleHeaderProps;
