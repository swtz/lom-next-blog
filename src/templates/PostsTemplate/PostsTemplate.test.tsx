import { renderTheme } from '../../styles/render-theme';
import { PostsTemplate, PostsTemplateProps } from '.';
import mock from './mock';

const props: PostsTemplateProps = mock;

describe('<PostsTemplate />', () => {
  it('should match snapshot', () => {
    const { container } = renderTheme(<PostsTemplate {...props} />);

    expect(container).toMatchSnapshot();
  });
});
