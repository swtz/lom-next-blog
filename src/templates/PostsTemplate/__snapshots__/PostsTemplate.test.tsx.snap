// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsTemplate /> should match snapshot 1`] = `
.c6 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c12 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c17 {
  text-align: center;
}

.c17 .c5 {
  margin: 0;
  margin-top: .8rem;
  margin-bottom: .8rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c17 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c17:hover .c5 {
  color: #DC143C;
}

.c17:hover img {
  opacity: 0.8;
}

.c18 {
  max-width: 100%;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.c14 {
  width: 100%;
  padding-left: 1.6rem;
}

.c15 {
  max-width: 120rem;
  margin: 0 auto;
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-columns: repeat(auto-fill,minmax(30rem,1fr));
  gap: 3.2rem;
  padding: calc(.8rem + 0.2rem);
}

.c15 .c16 {
  padding: 1rem;
  border: 0.5rem solid #DC143C;
  border-radius: 5%;
  height: 100%;
  -webkit-transition: border-color 300ms ease-in-out;
  transition: border-color 300ms ease-in-out;
}

.c15 .c16:hover {
  border-color: #333333;
}

.c20 {
  text-align: center;
  border-top: 0.1rem solid #DDDDDD;
}

.c20 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c21 {
  font-size: calc(1.6rem + 0.6rem);
  line-height: 1.5;
}

.c21 p {
  margin: 2.4rem 0;
}

.c21 a,
.c21 a:visited,
.c21 a:link {
  color: #DC143C;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c21 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c21 code {
  font-family: monospace;
  color: #DC143C;
  font-size: 1.6rem;
  background: #DDDDDD;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c21 pre {
  background: #000000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 2.4rem 0;
  font-size: 1.6rem;
  width: 100%;
  overflow-x: auto;
}

.c21 pre code {
  color: inherit;
  background: inherit;
}

.c21 img {
  max-width: 100%;
}

.c21 .image {
  background: #DDDDDD;
  line-height: 0;
  margin: 2.4rem 0;
}

.c21 .image figcaption {
  background: #DDDDDD;
  font-size: 1.6rem;
  padding: .8rem;
  text-align: center;
  line-height: 1.3;
}

.c21 .image-style-side {
  float: right;
  max-width: 50%;
  margin: 2.4rem .8rem;
}

.c21 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c21 ul,
.c21 ol {
  margin: 2.4rem 4.0rem;
}

.c21 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c21 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
}

.c21 table td,
.c21 table th {
  padding: .8rem;
  border: 0.1rem solid #DDDDDD;
}

.c21 .media iframe {
  width: 100%;
  height: auto;
  aspect-ratio: 16 / 9;
}

.c21 blockquote {
  border-left: 0.5rem solid #DC143C;
  color: #AAAAAA;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

.c22 {
  position: fixed;
  background: #000000;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  bottom: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  z-index: 6;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 > img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 0 3.2rem 0;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  font-size: 1.6rem;
  margin: 0 auto;
}

.c10 .c5 {
  margin: 0 0 calc(1.6rem - 1rem);
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  margin-left: 2.4rem;
  margin-right: 2.4rem;
  max-width: 48rem;
}

.c2 {
  background: #000000;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  left: -30rem;
  overflow-y: hidden;
}

.c3 {
  margin: auto;
  width: 100%;
}

.c4 .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.0rem;
}

.c4 .c5 img {
  border: 0.5rem solid #FFFFFF;
}

.c0 {
  position: fixed;
  top: 2.4rem;
  color: #FFFFFF;
  background: #000000;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  left: 26rem;
  -webkit-transition: all 350ms ease-in-out;
  transition: all 350ms ease-in-out;
  left: 1rem;
  color: #FFFFFF;
}

.c8 {
  display: block;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: .8rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #000000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  border-right: 0.5rem solid #DC143C;
  color: #DC143C;
}

.c9 {
  margin-top: 6.4rem;
}

.c13 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  margin-bottom: 6.4rem;
}

.c19 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  margin-bottom: 3.2rem;
}

@media (max-width:768px) {
  .c14 {
    padding-left: 0;
  }
}

@media (max-width:768px) {
  .c21 {
    font-size: calc(1.6rem + 0.2rem);
  }

  .c21 pre {
    font-size: 1.2rem;
  }

  .c21 .image-style-side {
    float: none;
    max-width: 100%;
    margin: 4.0rem 0;
  }
}

@media (max-width:550px) {
  .c10 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    text-align: center;
  }

  .c10 > div .c5 {
    margin: 1.6rem 0 calc(.8rem - 1rem);
  }
}

@media (max-width:768px) {
  .c2 {
    left: -32rem;
  }
}

<div>
  <div
    class=""
  >
    <a
      aria-label="Open or close menu"
      class="c0"
      href="#"
      title="Open or close menu"
    >
      <svg
        aria-hidden="true"
        aria-label="Open menu"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </a>
    <div
      aria-hidden="true"
      class="c2"
    >
      <nav
        class="c3"
      >
        <div
          class="c4"
        >
          <h1
            class="c5 c6"
          >
            <a
              class="c7"
              href="/"
              target="_self"
            >
              <img
                alt="Mundo do Arroz"
                src="https://pub-aea127bc483c40dea7b9339692296909.r2.dev/rice_icon_762798b97d.svg"
              />
            </a>
          </h1>
        </div>
        <a
          class="c8"
          href="#"
          target="_self"
        >
          home
        </a>
        <a
          class="c8"
          href="#page1"
          target="_self"
        >
          page1
        </a>
        <a
          class="c8"
          href="#page2"
          target="_self"
        >
          page2
        </a>
      </nav>
    </div>
    <div
      class="c9"
    >
      <header
        class="c10"
      >
        <h1
          class="c5 c6"
        >
          <a
            class="c7"
            href="/"
            target="_self"
          >
            <img
              alt="Mundo do Arroz - Um pedacinho do mundo que circunda um dos maiores carros-chefe da alimentação do PLANETA INTEIRO: o arroz."
              src="https://pub-aea127bc483c40dea7b9339692296909.r2.dev/rice_icon_762798b97d.svg"
            />
          </a>
        </h1>
        <div
          class="c11"
        >
          <h2
            class="c5 c12"
          >
            Mundo do Arroz
          </h2>
          <p>
            Um pedacinho do mundo que circunda um dos maiores carros-chefe da alimentação do PLANETA INTEIRO: o arroz.
          </p>
        </div>
      </header>
    </div>
    <div
      class="c13"
    >
      <div
        class="c14"
      >
        <div
          class="c15"
        >
          <div
            class="c16 c17"
          >
            <a
              href="/post/hello-rice"
            >
              <img
                alt="a fox holding a box containing a bottle and ice cubes"
                class="c18"
                src="https://pub-aea127bc483c40dea7b9339692296909.r2.dev/stages_of_rice_milling_8cb950424c.jpeg"
              />
            </a>
            <a
              href="/post/hello-rice"
            >
              <h2
                class="c5 c12"
              >
                Hello, rice!
              </h2>
            </a>
            <p
              class=""
            >
              Rice is a cereal grain and in its domesticated form is the staple food of over half of the world's population, particularly in Asia and Africa. Rice is the seed of the grass species Oryza sativa (Asian rice).
            </p>
          </div>
          <div
            class="c16 c17"
          >
            <a
              href="/post/hello-fox"
            >
              <img
                alt="a fox holding a box containing a bottle and ice cubes"
                class="c18"
                src="https://pub-aea127bc483c40dea7b9339692296909.r2.dev/DALL_E_2024_10_26_12_58_20_A_modern_logo_design_for_Deck_da_Praia_Conveniencia_featuring_a_fox_mascot_representing_cold_beer_delivery_The_fox_should_appear_in_motion_with_sp_fd834176b6.webp"
              />
            </a>
            <a
              href="/post/hello-fox"
            >
              <h2
                class="c5 c12"
              >
                Hello, fox!
              </h2>
            </a>
            <p
              class=""
            >
              Foxes are small-to-medium-sized omnivorous mammals belonging to several genera of the family Canidae.
            </p>
          </div>
          <div
            class="c16 c17"
          >
            <a
              href="/post/hello-bean"
            >
              <img
                alt="a fox holding a box containing a bottle and ice cubes"
                class="c18"
                src="https://pub-aea127bc483c40dea7b9339692296909.r2.dev/how_to_cook_dried_beans_6fb0921283.jpg"
              />
            </a>
            <a
              href="/post/hello-bean"
            >
              <h2
                class="c5 c12"
              >
                Hello, bean!
              </h2>
            </a>
            <p
              class=""
            >
              A bean is the seed of any plant in the legume family (Fabaceae) used as a vegetable for human consumption or animal feed.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c19"
    >
      <footer
        class="c20"
      >
        <div
          class="c21"
        >
          <p>
            Testando o CKEditor... and....
          </p>
        </div>
      </footer>
    </div>
    <a
      aria-label="Go to top"
      class="c22"
      href="#"
      title="Go to top"
    >
      <svg
        aria-hidden="true"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
        />
      </svg>
    </a>
  </div>
</div>
`;
